### PR TITLE
Platform API Demo in Java Adapter

### PR DESCRIPTION
**Pull request by weiteho**

### CODE REVIEW

File: java/platform-api-demo/src/main/java/com/openfin/demo/PlatformApiDemo.java [hunk 1]
```diff
@@ -0,0 +1,879 @@
...
+			btnApply.setEnabled(p != null);
+			tfSelectedPlatformUuid.setText(p == null ? "N/A" : p.getUuid());
+		});
+
+		pnlSnapshot.add(pnlTop, BorderLayout.NORTH);
+		pnlSnapshot.add(pnlCenter, BorderLayout.CENTER);
+		return pnlSnapshot;
+	}
+
+	JPanel createLayoutPanel() {
```
Comment (rdepena) [inline java/platform-api-demo/src/main/java/com/openfin/demo/PlatformApiDemo.java:453]: Can we just store this in memory ?

Something like you name aLayout as you save it, goes to a list then you pick one and apply it.

File: java/platform-api-demo/src/main/java/com/openfin/demo/PlatformApiDemo.java [hunk 2]
```diff
@@ -0,0 +1,879 @@
...
+	JPanel createWindowPanel() {
+		JPanel pnl = new JPanel(new BorderLayout());
+		JPanel pnlWinOpts = new JPanel(new GridBagLayout());
+		pnlWinOpts.setBorder(BorderFactory.createTitledBorder("Window Options"));
+		GridBagConstraints gbConst = new GridBagConstraints();
+		gbConst.insets = new Insets(5, 5, 5, 5);
+		gbConst.fill = GridBagConstraints.BOTH;
+		gbConst.gridx = 0;
+		gbConst.gridy = 0;
+		JCheckBox cbInitLayout = new JCheckBox("Init Layout");
```
[the remark this answers is not in the document]
Comment (weiteho) [reply java/platform-api-demo/src/main/java/com/openfin/demo/PlatformApiDemo.java]: Changed to Platform Window per comment.

File: java/platform-api-demo/src/main/java/com/openfin/demo/PlatformApiDemo.java [hunk 3]
```diff
@@ -0,0 +1,879 @@
...
+		gbConst.gridx = 0;
+		gbConst.gridy = 0;
+		pnlWinOpts.add(new JLabel("Name"), gbConst);
+		gbConst.gridy++;
+		pnlWinOpts.add(new JLabel("URL"), gbConst);
+		gbConst.weightx = 1;
+		gbConst.gridwidth = 2;
+		gbConst.gridx = 1;
+		gbConst.gridy = 0;
+		JTextField tfName = new JTextField("viewName");
```
[the remark this answers is not in the document]
Comment (weiteho) [reply java/platform-api-demo/src/main/java/com/openfin/demo/PlatformApiDemo.java]: removed default name per comment

File: java/platform-api-demo/src/main/java/com/openfin/demo/PlatformApiDemo.java [hunk 4]
```diff
@@ -0,0 +1,879 @@
...
+		if (selectionPath != null) {
+			DefaultMutableTreeNode node = (DefaultMutableTreeNode) selectionPath.getLastPathComponent();
+			if (clazz.isInstance(node.getUserObject())) {
+				n = (T) node.getUserObject();
+			}
+		}
+		return n;
+	}
+
+	JPanel createPlatformStartFromManifestPanel() {
```
[the remark this answers is not in the document]
Comment (weiteho) [reply java/platform-api-demo/src/main/java/com/openfin/demo/PlatformApiDemo.java:144]: added default view url option and checked box to create a view when starting the platform.

File: java/platform-api-demo/src/main/java/com/openfin/demo/PlatformApiDemo.java [hunk 5]
```diff
@@ -0,0 +1,879 @@
...
+		p.setBorder(BorderFactory.createEmptyBorder(10, 10, 10, 10));
+		p.setLayout(new BoxLayout(p, BoxLayout.Y_AXIS));
+		p.add(createPlatformStartFromManifestPanel());
+		p.add(createPlatformStartPanel());
+		p.add(new Box.Filler(new Dimension(0, 0), new Dimension(0, Short.MAX_VALUE),
+				new Dimension(0, Short.MAX_VALUE)));
+		return p;
+	}
+
+	JPanel createWindowPanel() {
```
[the remark this answers is not in the document]
Comment (weiteho) [reply java/platform-api-demo/src/main/java/com/openfin/demo/PlatformApiDemo.java:233]: removed default view name per comment.

File: java/platform-api-demo/src/main/java/com/openfin/demo/PlatformApiDemo.java (hunk 1 above)
Comment (weiteho) [reply java/platform-api-demo/src/main/java/com/openfin/demo/PlatformApiDemo.java:481]: changed the implementation per comment.

File: java/platform-api-demo/src/main/java/com/openfin/demo/PlatformApiDemo.java [hunk 6]
```diff
@@ -0,0 +1,879 @@
...
+		pnlSave.setLayout(new BoxLayout(pnlSave, BoxLayout.X_AXIS));
+		pnlSave.setBorder(BorderFactory.createTitledBorder("Save Snapshot"));
+		JTextField tfSavePath = new JTextField(new File("snapshot.json").getAbsolutePath().toString());
+		tfSavePath.setPreferredSize(new Dimension(Short.MAX_VALUE, tfSavePath.getPreferredSize().height));
+		JButton btnSave = new JButton("Save...");
+		btnSave.setEnabled(false);
+		btnSave.addActionListener(e -> {
+			Platform platform = this.getSelectedNode(Platform.class);
+			if (platform != null) {
+				JFileChooser fileChooser = new JFileChooser(new File(".").getAbsoluteFile());
```
[the remark this answers is not in the document]
Comment (weiteho) [reply java/platform-api-demo/src/main/java/com/openfin/demo/PlatformApiDemo.java]: changed implementation per comment.

**Comment by weiteho** [comment]
updated demo app per comments.

**Pull request merged**